mythril/analysis/solver.py

```diff
@@ -122,12 +125,59 @@ def get_transaction_sequence(
         ).as_long()
 
     concrete_initial_state = _get_concrete_state(initial_accounts, min_price_dict)
-
+    if isinstance(transaction_sequence[0], ContractCreationTransaction):
+        code = transaction_sequence[0].code
+        _replace_with_actual_sha(concrete_transactions, model, code)
+    else:
+        _replace_with_actual_sha(concrete_transactions, model)
     steps = {"initialState": concrete_initial_state, "steps": concrete_transactions}
 
     return steps
 
 
+def _replace_with_actual_sha(
+    concrete_transactions: List[Dict[str, str]], model: z3.Model, code=None
+):
+    for tx in concrete_transactions:
+        if hash_matcher not in tx["input"]:
+            continue
+        if code is not None and code.bytecode in tx["input"]:
+            s_index = len(code.bytecode) + 2
+        else:
+            s_index = 10
+        for i in range(s_index, len(tx["input"])):
```

Comment:
I am not putting in `len(tx["input"]) - 64 or 63 or whatever` because it is a bit hard to understand it and, so I am just checking data_slice == 64 in the loop which is easier to read.